faithcomesbyhearing/dbp
# FCBH-407/408 Backend: Plans Support

# Description
- Created PlansController for CRUD endpoints
- Created start plan endpoint
- Updated PlaylistController to show the author
- Added `external_content` column to playlist
- Created Plan/PlanDay/UserPlan Models
- Installed `spatie/eloquent-sortable` to add the ability to sort tables
- Installed migrations
- Updated documentation

## Issue Link
Original Story: [FCBH-407](https://fullstacklabs.atlassian.net/browse/FCBH-407) 
Original Story: [FCBH-408](https://fullstacklabs.atlassian.net/browse/FCBH-408) 
Review Story: [FCBH-562](https://fullstacklabs.atlassian.net/browse/FCBH-562) 
Review Story: [FCBH-563](https://fullstacklabs.atlassian.net/browse/FCBH-563) 

## How Do I QA This

- On your local environment run `php artisan migrate --database="dbp_users"` to generate the database tables
- On your local environment run `composer install`
- On your local environment run `http://dbp.test/open-api-v4.json` and verify that now create, read and update endpoints are documented
- Test that now you can create, read and update a plan
- Plans items will be addressed on FCBH-454

## Checklist
